src/Sparrow.Server/Utils/VxSort/codegen/bitonic_avx2.py

```diff
@@ -0,0 +1,652 @@
+        public static void sort_01v_{suffix}({g.generate_param_def_list(1)}) {{
+            {self.generate_1v_sort(ascending, 1)}        
+        }}"""
+
+        print(s, file=f)
+
+    def generate_1v_merge(self, ascending, start):
+        g = self
+        type = self.type
+        var = self._var_names
+        maybe_cmp = lambda: ", cmp" if (type == "long" or type == "ulong") else ""
+        maybe_topbit = lambda: f"{g.vector_type()} topBit = Vector256.Create(1UL << 63);" if type == "ulong" else ""
+
+        s = f"""
+            {g.vector_type()}  min, max, s{maybe_cmp()};
+            {maybe_topbit()}"""
+        if g.vector_size() == 8:
+            s += f"""
+            s = {g.generate_cross(var[start])};
+            min = {g.generate_min("s", var[start])};
+            max = {g.generate_max("s", var[start])};
+            {var[start]} = {g.generate_blend_B4("min", "max", ascending)};"""
+
+        s += f"""
+            s = {g.generate_shuffle_X2(var[start])};
+            {g.crappity_crap_crap("s", var[start])}
+            min = {g.generate_min("s", var[start])};
+            max = {g.generate_max("s", var[start])};
+            {var[start]} = {g.generate_blend_B2("min", "max", ascending)};
+
+            s = {g.generate_shuffle_X1(var[start])};
+            {g.crappity_crap_crap("s", var[start])}
+            min = {g.generate_min("s", var[start])};
+            max = {g.generate_max("s", var[start])};
+            {var[start]} = {g.generate_blend_B1("min", "max", ascending)};"""
+
+        return s
+
+    def generate_1v_merge_sorters(self, f, ascending: bool):
+        g = self
+        suffix = "ascending" if ascending else "descending"
+
+        s = f"""
+        [MethodImpl(MethodImplOptions.AggressiveInlining | MethodImplOptions.AggressiveOptimization)]
+        public static void sort_01v_merge_{suffix}({g.generate_param_def_list(1)}) {{
+            {g.generate_1v_merge(ascending, 1)}
+        }}
+        """
+
+        print(s, file=f)
+
+    def generate_compounded_merge(self, width, ascending, start, numParams=1):
+        g = self
+        suffix = "ascending" if ascending else "descending"
+
+        if width == 1:
+            return f"""
+        {{        
+            //sort_{1:02d}v_merge_{suffix}({g.generate_param_list(1, start)});      
+            {g.generate_1v_merge(ascending, start)}
+        }}"""
+        elif width < g.unroll_bitonic_sorters:
+            return f"""
+        {{
+            //sort_{width:02d}v_merge_{suffix}({g.generate_param_list(start, numParams)}); 
+            {g.generate_xv_compounded_merger(width, ascending, start)}
+        }}"""
+        else:
+            return f"""
+        {{
+            //sort_{width:02d}v_merge_{suffix}({g.generate_param_list(start, numParams)});
+            sort_{width:02d}v_merge_{suffix}({g.generate_param_list(start, numParams)});
+        }}"""
+
+    def generate_compounded_sort(self, width, ascending, start, numParams=1):
+        g = self
```

Comment:
@damageboy Here I will essentially allow to do the recursive unrolling of the bitonic sorters to avoid having method calls. You generate essentially a wall of primitive instructions instead of method calls.